src/embedded-checkout/embedded-checkout.ts

```diff
@@ -0,0 +1,84 @@
+import { IFrameComponent } from 'iframe-resizer';
+
+import { EmbeddedCheckoutEventMap, EmbeddedCheckoutEventType } from './embedded-checkout-events';
+import EmbeddedCheckoutListener from './embedded-checkout-listener';
+import EmbeddedCheckoutOptions from './embedded-checkout-options';
+import ResiableIframeCreator from './resizable-iframe-creator';
+
+export default class EmbeddedCheckout {
+    private _iframe?: IFrameComponent;
+    private _isAttached: boolean;
+
+    /**
+     * @internal
+     */
+    constructor(
+        private _iframeCreator: ResiableIframeCreator,
+        private _messageListener: EmbeddedCheckoutListener,
+        private _options: EmbeddedCheckoutOptions
+    ) {
+        this._isAttached = false;
+
+        if (this._options.onComplete) {
+            this.on(EmbeddedCheckoutEventType.CheckoutComplete, this._options.onComplete);
+        }
+
+        if (this._options.onError) {
+            this.on(EmbeddedCheckoutEventType.CheckoutError, this._options.onError);
+        }
+
+        if (this._options.onLoad) {
+            this.on(EmbeddedCheckoutEventType.CheckoutLoaded, this._options.onLoad);
+        }
+
+        if (this._options.onFrameLoad) {
+            this.on(EmbeddedCheckoutEventType.FrameLoaded, this._options.onFrameLoad);
+        }
+    }
+
+    attach(): Promise<this> {
+        if (this._isAttached) {
+            return Promise.resolve(this);
+        }
+
+        this._isAttached = true;
```

Comment:
shouldn't this be set on success?  I guess is not a big deal, as if it fails you can always detach and attach again